projects/igniteui-angular/src/lib/select/select-item.component.ts

```diff
@@ -53,7 +53,8 @@ export class IgxSelectItemComponent extends IgxDropDownItemComponent implements
     }
 
     public set selected(value: any) {
-        if (value && !this.isHeader && !this.disabled) {
+        if (value && !this.isHeader && !this.disabled) { // TODO: Refactor: inherited isHeader is regarding group items
```

Comment:
`isHeader`, in context of `IgxDropdownItem`, is item which has a role of header. To group items you can use `IgxDropDownGroup` again in the context of items. And finally you can add header and footer to entire list, which is not in the context of items. You can remove this comment as there is no TODO here - header items should not be selected, and `isHeader` is reasonable in the context of item.